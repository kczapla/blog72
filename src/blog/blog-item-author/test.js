import React from 'react'
import renderer from 'react-test-renderer'

import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'

import BlogItemAuthor from './index'

describe('blog item author', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<BlogItemAuthor content={'tmp'}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('redirects to an author', () => {
    const hrefValue = 'https://google.com'
    render(<BlogItemAuthor href={hrefValue} name={'test'}/>)
    expect(screen.getByRole('link')).toHaveAttribute('href', hrefValue)
  })

  it('has name', () => {
    render(<BlogItemAuthor href={'tmp'} name={'test'}/>)
    expect(screen.getByRole('link')).toHaveTextContent('test')
  })
})
