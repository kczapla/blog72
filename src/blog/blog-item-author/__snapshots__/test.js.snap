// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog item author renders correctly 1`] = `
<span
  className="sc-bdfBwQ kGqeyI"
>
  <a
    className="sc-gsTCUz elSzKW"
  />
</span>
`;
